assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -11742,6 +11747,53 @@ else if (typeOfGrade == Assignment.CHECK_GRADE_TYPE) {
 				}
 			}
 		}
+		
+		
+		public String getGradeForUserInGradeBook(String userId)
+		{
+			String rv = null;
+			Assignment m = getAssignment();
+			String gAssignmentName = StringUtils.trimToNull(m.getProperties().getProperty(AssignmentService.PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT));
+			if (gAssignmentName != null)
+			{
+				GradebookService g = (GradebookService)  ComponentManager.get("org.sakaiproject.service.gradebook.GradebookService");
+				String gradebookUid = m.getContext();
+				SecurityAdvisor securityAdvisor = new SecurityAdvisor() {
+					public SecurityAdvice isAllowed(String userId, String function,
```

Comment:
What's the purpose of having the security advisor here? What is this giving this user access to that they otherwise normally wouldn't have?